File: Giovanna-e-Gabriela/CPP/Lista-1/ex-09.cpp
/*
Data: 07/05/2026
Autores: Gabriela Onaga e Giovanna Costa
Descrição: Ler um número entre 50 e 100. Exibir a sequência de Fibonacci até esse número.
*/

#include <iostream>

using namespace std;

int main() {
    int limite;

    cout << "Digite um número entre 50 e 100: ";
    cin >> limite;

    if (limite < 50 || limite > 100) {
        cout << "Número inválido!" << "\n\n";
    } else {
        int numeroA = 0;
        int numeroB = 1;

        cout << "Sequência de Fibonacci:" << "\n\n";

        while (numeroA <= limite) {
            cout << numeroA << " ";

            int proximo = a + b;
            numeroA = numeroB;
            numeroB = proximo;
        }
    }

    return 0;
}

File: Giovanna-e-Gabriela/Python/python-conversion/python/ex09.py
"""
    Autor: Gabriela Onaga e Giovanna Costa
    Data: 11/08/2026
    Descrição: Calcula o fatorial de um número inteiro positivo.
"""

numero = int(input("Digite um numero inteiro positivo: "))

contador = 1
fatorial = 1

if numero >= 0:
    while contador <= numero:
        fatorial = fatorial * contador
        contador += 1

    print("Fatorial:", fatorial)

else:
    print("Numero invalido.")
